### PR TITLE
update doc in build section

### PR DESCRIPTION
**Pull request by mszhanyi**
Why:
To keep VS version always updated in README
1. update VS version link in CI. It's more convenient for my PR robot to update the version in README once the VS in CI is updated. and permlink isn't stable.
2. Move `building on legacy code` to development tips. The table is big and it looks the REAMD not updated at the first sight.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,6 +33,7 @@
 - [CUDA development tips](#cuda-development-tips)
 - [Windows development tips](#windows-development-tips)
   - [Known MSVC (and MSVC with NVCC) bugs](#known-msvc-and-msvc-with-nvcc-bugs)
+  - [Building on legacy code and CUDA](#build-legacy-code)
```
Comment (samestep) [inline CONTRIBUTING.md:36]: I think this is incorrect:
```suggestion
  - [Building on legacy code and CUDA](#building-on-legacy-code-and-cuda)
```

File: README.md [hunk 2]
```diff
@@ -267,9 +267,11 @@ On Windows
 Choose Correct Visual Studio Version.
 
 Visual Studio upgrades are very often. Sometimes, there're regressions in some new versions.
-It'd best to use the same Visual Studio Version as [PyTorch CI's](https://github.com/pytorch/pytorch/blob/a1bd7918cc5a06cbef6c5178259bf0a7b5ab1ce3/.circleci/scripts/vs_install.ps1#L4).
+It'd best to use the same Visual Studio Version [16.8.5](https://github.com/pytorch/pytorch/blob/master/.circleci/scripts/vs_install.ps1) as PyTorch CI's.
 You can use Visual Studio Enterprise, Professional or Community though PyTorch CI uses Visual Studio BuildTools.
 
+If you want to build legacy python code, please refert to [Building on legacy code and CUDA](https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#build-legacy-code)
```
Comment (samestep) [inline README.md:273]: Same here:
```suggestion
If you want to build legacy python code, please refert to [Building on legacy code and CUDA](https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#building-on-legacy-code-and-cuda)
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (mszhanyi) [reply CONTRIBUTING.md]: Thank you very much! fixed

**Comment by mszhanyi** [comment]
@samestep any more comments

**Comment by samestep** [comment]
> @samestep any more comments

@mszhanyi ah my apologies! I forgot about this PR... I actually don't have the context to review the main change, I was just commenting on the tables of contents; does someone else have the context to review this? @pytorch/pytorch-dev-infra

**Review by samestep — APPROVED** [review]
thank you!

**Review by janeyx99 — APPROVED** [review]

**Pull request closed without merging**